src/python/pants/testutil/rule_runner.py

```diff
@@ -656,89 +659,163 @@ class MockGet(Generic[_O]):
 
 @dataclass(frozen=True)
 class MockRequestExceptionComparable:
-    category: Literal["Get"] | Literal["Effect"] | str
-    output_type: str | None
-    input_types: tuple[str, ...]
+    category: Literal["Get", "Effect"]
+    output_type: type | None
+    input_types: tuple[type, ...]
 
     def __str__(self):
-        inputs = ",".join(str(e) for e in self.input_types)
```

Comment:
Casting what was already typed as a `str` into a `str` shows that this code was very existentially confused. I switched it to use `type`s and not their stringifications, for clarity (and to avoid confusion elsewhere where we compare types).